[refs #00224] Fix double heading border

Android63 opened the pull request:
Following an earlier update, the faux bottom border was no longer being
displayed. I investigated and it was being displayed as a table so I changed it to block.

Android63 committed:
[refs #00224] Fix double heading border

Following an earlier update, the faux bottom border was no longer being
displayed.

cehwitham commented:
Was there a reason for it being displayed as a table initially?

It used to work, so it would be good to understand what broke it. Anything in the header layout changes we made?

Android63 commented:
I'll look into it @cehwitham

Android63 commented:
@csswizardry could you take a look at this please? I believe the issue occurred after you updated tools/_tools.mixins.scss to use `table` rather than `block` when there's both a `before` and `after` pseudo element.  See https://github.com/NHSLeadership/nightingale/commit/4d4f48acdc1375e4d3622e5c2b861249442325bc

Android63 committed:
Merge branch 'develop' into tkt-00224

Android63 committed:
[refs #00224] Mark fix as temporary

Add comment pending review.

Android63 committed:
[refs #00224] Fix ribbon & breadcrumb top margins

Prevent faux double header taking up top margin of ribbon or
breadcrumb beneath it.

Android63 commented:
I've marked my fix and raised an issue to get Harry to review it. While I was here I fixed the issue that the faux double border was overwriting the ribbon or breadcrumbs directly below it.

This shows the page-specific ribbon being overwritten by the faux border:

![screen shot 2018-05-11 at 12 05 04](https://user-images.githubusercontent.com/1991226/39921857-661a0ec8-5514-11e8-92ef-34c5b2b5d2f9.png)

This is the result of increasing the top margin on the ribbon to compensate:

![screen shot 2018-05-11 at 12 05 44](https://user-images.githubusercontent.com/1991226/39921889-89c08474-5514-11e8-987d-f78037b1b236.png)

This shows the same problem occurring with breadcrumbs:

![screen shot 2018-05-11 at 12 06 12](https://user-images.githubusercontent.com/1991226/39921930-b0e4eda6-5514-11e8-870d-5c00c9f57753.png)

This is the result of increasing the margin for breadcrumbs:

![screen shot 2018-05-11 at 12 06 49](https://user-images.githubusercontent.com/1991226/39922005-f53c4030-5514-11e8-80e8-5739061214d3.png)

This shows that increasing the breadcrumb margin causes a gap to appear when it sits under a ribbon:

![screen shot 2018-05-11 at 12 18 30](https://user-images.githubusercontent.com/1991226/39922141-a249a574-5515-11e8-8d5d-b24ddcfb7529.png)

This is the result of using CSS sibling relationships to apply a larger negative top margin when a ribbon is present above the breadcrumbs:

![screen shot 2018-05-11 at 12 07 13](https://user-images.githubusercontent.com/1991226/39922032-1005b9e6-5515-11e8-9460-a396fae74339.png)

Android63 commented:
@cehwitham could you review this please?
